kong/plugins/prometheus/exporter.lua

```diff
@@ -168,6 +174,50 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- upstream targets accessible?
+  local balancer = require "kong.runloop.balancer"
+  local upstreams_dict = balancer.get_all_upstreams()
+  for upstream_name, upstream_id in pairs(upstreams_dict) do
+    -- same logic from kong.db.dao.targets
+    local health_info
+    health_info, err = balancer.get_upstream_health(upstream_id)
+    if err then
+      ngx.log(ngx.ERR, "failed getting upstream health: ", err)
+    end
+
+    if health_info then
+      local status = 0 -- healthchecks is off
+      for target_name, target_info in pairs(health_info) do
+        if target_info ~= nil and
+          #target_info.addresses > 0 then
```

Comment:
let's be protective here, also check for nil-ness of `target_info.addresses` before getting its length

Reply:
this code is a copy from kong.db.dao.targets, it would be nice if the dao.targets provides it right the way